superset/assets/visualizations/nvd3_vis.js

```diff
@@ -97,9 +100,22 @@ function nvd3Vis(slice, payload) {
   const isExplore = $('#explore-container').length === 1;
 
   let data;
+  // Get the most significant part of the series name (key)
+  const mostSignificant = payload.data
+    .filter(x => Array.isArray(x.key) && x.key.length > 1)
+    .map(x => x.key[0]);
+  // # of distinct most significant keys
+  const distinctMostSignificant = mostSignificant
+    .reduce((distinct, v) => distinct.add(v), new Set()).size;
+  // remove the msg key from the label if these conditions are true
+  const removeMostSignificant = mostSignificant.length === payload.data.length &&
+    distinctMostSignificant === 1;
```

Comment:
instead of this (complicated) logic we could simplify  how we set the `removeMostSignificant` flag to:
```
const removeMostSignificant = slice.formData.metrics.length === 1;
```
However that would make the label generation depend on the fromData and it would therefore be less generic.